contracts/src/spaces/facets/membership/metadata/MembershipMetadata.sol

```diff
@@ -13,6 +14,16 @@ import {ERC721ABase} from "contracts/src/diamond/facets/token/ERC721A/ERC721ABas
 import {Facet} from "contracts/src/diamond/facets/Facet.sol";
 
 contract MembershipMetadata is ERC721ABase, Facet {
+  /// @dev This event emits when the metadata of a token is changed.
+  /// So that the third-party platforms such as NFT market could
+  /// timely update the images and related attributes of the NFT.
+  event MetadataUpdate(uint256 _tokenId);
+
+  /// @notice Emits an event to trigger metadata refresh when the space info is updated
+  function refreshMetadata() external {
+    emit MetadataUpdate(type(uint256).max);
```

Comment:
I dont think this is right - we need BatchMetadataUpdate

event BatchMetadataUpdate(uint256 _fromTokenId, uint256 _toTokenId);

To refresh a whole collection, emit _toTokenId with type(uint256).max

https://docs.github.com/en/get-started/accessibility/keyboard-shortcuts#site-wide-shortcuts

Reply:
Good catch. Thanks.